src/app/space/forge-wizard/forge.service.ts

```diff
@@ -0,0 +1,64 @@
+
+import { Injectable, Inject } from '@angular/core';
+import { Http, RequestOptions, Headers } from '@angular/http';
+import { FABRIC8_FORGE_API_URL } from 'a-runtime-console';
+import { State } from 'app/space/forge-wizard/state.component';
+import { AuthenticationService } from 'ngx-login-client';
+import { Gui } from './gui.model';
+import { History } from './history.component';
+
+@Injectable()
+export class ForgeService {
+
+  constructor(private http: Http,
+    private auth: AuthenticationService,
+    @Inject(FABRIC8_FORGE_API_URL) private apiUrl: string) {
+      this.apiUrl += '/forge';
+  }
+
+  commandInfo(command: string): Promise<Gui> {
+    let options = new Headers({ 'Authorization': 'Bearer ' + this.auth.getToken() });
+    return this.http.get(`${this.apiUrl}/commands/${command}`, new RequestOptions({ headers: options }))
+      .retryWhen(errors => errors.delay(3000).scan((acc, source, index) => {
```

Comment:
I think `retryWhen` is a part of RxJS 4. We are using 5. I do not think this will work.

Reply:
no this works, I do the same on launch

Reply:
Coming from https://github.com/ReactiveX/rxjs/blob/5.2.0/src/operator/retryWhen.ts